### PR TITLE
Gke spot fix

### PR DESCRIPTION
**Pull request by varnastadeus**

### CODE REVIEW

File: handler/gcp_test.go [hunk 1]
```diff
@@ -4,25 +4,33 @@ import (
...
 	"github.com/stretchr/testify/require"
 )
 
 func TestGCPInterruptChecker(t *testing.T) {
-	s := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		require.Equal(t, "/computeMetadata/v1/instance/preempted", r.URL.String())
-
-		w.WriteHeader(http.StatusOK)
-		_, err := w.Write([]byte("TRUE"))
+	mux := http.NewServeMux()
```
Comment (viktorasm) [inline handler/gcp_test.go:16]: nitpick: would have provided mocked metadata client with a mocked `metadata.Get` method instead; by building HTTP server we're testing google SDK which is not desirable - tests break if client implementation changes, e.g. requires additional http header.

File: handler/handler.go [hunk 2]
```diff
@@ -47,29 +49,35 @@ func NewSpotHandler(
...
+		gracePeriod:      30 * time.Second,
 	}
 }
 
 func (g *SpotHandler) Run(ctx context.Context) error {
 	t := time.NewTicker(g.pollWaitInterval)
 	defer t.Stop()
 
+	var once sync.Once
+	deadline := time.NewTimer(24 * 365 * time.Hour)
```
Comment (viktorasm) [inline handler/handler.go:61]: why is this needed?

**Review by anjmao — APPROVED** [review]

File: handler/handler.go (hunk 2 above)
Comment (varnastadeus) [reply handler/handler.go:61]: this is a simple way to keep checking for preemption even if we received a termination signal, and when deadline actually occurs we exit the application
